spec/app_spec.rb

```diff
@@ -0,0 +1,118 @@
+require 'spec_helper'
+require 'rack/test'
+require_relative '../app'
+
+def reset_table
+  seed_sql = File.read('spec/test_seeds.sql')
+  connection = PG.connect({ host: '127.0.0.1', dbname: 'chitter_test' })
+  connection.exec(seed_sql)
+end
+
+describe App do
+  # This is so we can use rack-test helper methods.
+  include Rack::Test::Methods
+
+  # We need to declare the `app` value by instantiating the Application
+  # class so our tests work.
+  let(:app) { App.new }
+  before(:all) { reset_table }
+  after(:all) { reset_table }
+  after(:each) { expect(@request.status).to eq 200 }
+
+  context 'GET /' do
+    it 'returns status 200 and index page' do
+      @request = get('/')
+      expect(@request.body).to include('Posts')
+    end
+  end
+
+  context 'GET /login' do
+    it 'returns a login form' do
+      @request = get('/login')
+      expect(@request.body).to include('Log in')
+    end
+  end
+
+  context 'POST /login' do
+    it 'returns a customised homepage' do
+      @request = post('/login', username: 'username2', password: 'password2')
+      expect(@request.body).to include('Hello Jimmy Kimmel')
+    end
+  end
+
+  context 'POST /logout' do
+    it 'logs out and returns to index' do
+      @request = post('/logout/2')
+      expect(@request.body).to include('Posts')
```

Comment:
Could maybe add more test conditions here? Are there any pre-requisites to logout?